### ISSUE TITLE
Require PRs and passing CI for commits to this repository

### ISSUE THREAD

**Issue by MikeMcQuaid** [Homebrew/brew-pip-audit#84]
There's currently a bot that commits directly which means these branch protections security features cannot be enabled for this repository.

It would be ideal to have the bot open PRs automatically or push the data somewhere other than this repository.

**Comment by MikeMcQuaid** [comment]
For example, in this case the regularly regenerated files could instead be deployed to GitHub Pages instead.

**Comment by MikeMcQuaid** [comment]
@woodruffw @alex Would love to help move this along somehow if I could. The two ways that jump out to me (based on other Homebrew ecosystem things) are:
1. have the bot create PRs that a human eyeballs and merges instead of pushing direct to `main`
2. either this repository or another one instead deploys these JSON files to GitHub Pages which `brew-pip-audit` then consumes instead of needing these files committed into this repository

There may also be other approaches here that I don't currently see (e.g. plain old GitHub Actions caching of this data). I'm happy to help move forwards whatever approach you think makes most sense here. Right now this is one of the 2 remaining repos that don't have human review for all changes and I'd love to fix that in the next 2-3 months 😁

**Comment by woodruffw** [comment]
> * have the bot create PRs that a human eyeballs and merges instead of pushing direct to `main`

I'm a fan of this 🙂 -- I _think_ nothing about our current flow would be badly disturbed/disrupted by changing to PRs over pushing directly.

I'm curious if @alex has strong feelings either way; the only "downside" to the PR flow is that a human has to merge it each day (or whatever window we choose), but I don't mind being the one to do that.

**Comment by woodruffw** [comment]
Yeah, it would be just a rubber-stamp effectively (there's also the annoying bit where machine-generated PRs don't trigger CI automatically unless they come from a PAT or app, although in this context I suppose we don't have that issue thanks to @BrewTestBot).

I suppose having it be a JSON or whatever dump on GitHub Pages would be the best of both worlds then -- no protection bypasses and no rubber-stamp for data-only changes.

**Comment by alex** [comment]
Yes. (It'd also give us a place to put some of the tables that we currently put in GHA output)

**Comment by MikeMcQuaid** [comment]
Thanks for input here too @alex, very helpful ❤

**Comment by woodruffw** [comment]
Working on this in #210!
